cpg_workflows/stages/rd_combiner.py

```diff
@@ -622,11 +692,21 @@ def queue_jobs(self, dataset: Dataset, inputs: StageInput) -> StageOutput:
 
         variant_mt = inputs.as_path(target=get_multicohort(), stage=AnnotateCohortSmallVariantsWithHailQuery)
 
+        if only_families := config_retrieve(['workflow', dataset.name, 'only_families'], []):
+            family_sg_ids = query_for_family_sequencing_groups(
+                dataset.name,
+                config_retrieve(['workflow', 'sequencing_type']),
+                only_families,
+            )['family_sg_ids']
+        else:
+            family_sg_ids = None
+
         # write a list of all the SG IDs to retain
         if not config_retrieve(['workflow', 'dry_run'], False):
             with outputs['id_file'].open('w') as f:
                 for sg in dataset.get_sequencing_groups():
-                    f.write(f'{sg.id}\n')
+                    if not family_sg_ids or sg.id in family_sg_ids:
```

Comment:
```suggestion
                    if (family_sg_ids is None) or (sg.id in family_sg_ids):
```

It's possible with the current logic to provide a list of families, then fail to map them to any SGs (family_sg_ids is empty). In that case `not family_sg_ids` still evaluates to True, so we keep all SGs.

I think a warning (probably quitting pipeline setup) would be valid if we give a specific family subset and can't find any SGs for them, given that we tried to acquire a specific subset, but won't achieve that.